src/main/java/eu/adlogix/appnexus/oas/client/service/CampaignService.java

```diff
@@ -0,0 +1,111 @@
+package eu.adlogix.appnexus.oas.client.service;
+
+import static eu.adlogix.appnexus.oas.client.utils.ValidatorUtils.checkNotEmpty;
+import static eu.adlogix.appnexus.oas.client.utils.ValidatorUtils.checkNotNull;
+import static org.apache.commons.lang3.StringUtils.defaultIfEmpty;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import eu.adlogix.appnexus.oas.client.domain.Campaign;
+import eu.adlogix.appnexus.oas.client.domain.StatefulDomainManager;
+import eu.adlogix.appnexus.oas.client.domain.enums.BillTo;
+import eu.adlogix.appnexus.oas.client.domain.enums.CampaignType;
+import eu.adlogix.appnexus.oas.client.parser.XmlToCampaignParser;
+import eu.adlogix.appnexus.oas.client.transform.CampaignCreateParameterMapTransformer;
+import eu.adlogix.appnexus.oas.client.transform.CampaignParameterMapTransformer;
+import eu.adlogix.appnexus.oas.client.transform.CampaignUpdateParameterMapTransformer;
+import eu.adlogix.appnexus.oas.client.xml.ResponseParser;
+import eu.adlogix.appnexus.oas.client.xml.XmlRequestGenerator;
+
+/**
+ * Service Class which provides functions for all Campaign related operations
+ * 
+ */
+public class CampaignService extends AbstractOasService {
```

Comment:
Well there are a lot of discussions regarding this exact question :) Some say that if you don't have and will probably never have a second implementation, why would you need an interface at all. That being said, I'm not convinced that an interface shouldn't exist simply because only one implementation will exist. 

That being said, how do we call that 'only one' implementation. Some say 'InterfaceName' + Impl , but I really don't like the 'Impl' ... some others say Default + 'InterfaceName' .. and I tend to lean more on that solution.

What ya think?